test/integration-spec/helper.insights.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+var utils = require('../integration-utils.js');
+var setup = utils.setupSimple;
+
+var algoliasearchHelper = utils.isCIBrowser ? window.algoliasearchHelper : require('../../');
+
+var test = require('tape');
+var random = require('lodash/random');
+
+if (!utils.shouldRun) {
+  test = test.skip;
+}
+var indexName = '_travis-algoliasearch-helper-js-' +
+  (process.env.TRAVIS_BUILD_NUMBER || 'DEV') +
+  'helper_searchonce' + random(0, 5000);
+
+var dataset = [
+  {objectID: '1', name: 'TestName'}
+];
+
+var config = {};
+
+test(
+  '[INT][INSIGHTS] search with clickAnalytics should have a queryID',
+  function(t) {
+    setup(indexName, dataset, config).
+    then(function(client) {
+      var helper = algoliasearchHelper(client, indexName, {clickAnalytics: true});
+      helper.on('result', function(content) {
+        t.notEqual(content.queryID, undefined);
+        t.end();
```

Comment:
not sure, but could you say it needs to be a string of X length with `tape`?

Reply:
worst case, `typeof` and `.length`

Reply:
That's a good suggestion for improvement, can you do that @JonasBa ?

Reply:
@Haroenv You could also use a regex to test the string and the length 👍 

```js
/\S{2}/.test('ab') // === true
```

Reply:
However the length might change and be formatted differently. So maybe we should just test that it's a non empty string.